pkg/util/utils.go

```diff
@@ -79,6 +81,34 @@ func CreateKubeClient(apiserver string, kubeconfig string) (clientset.Interface,
 	return kubeClient, nil
 }
 
+// CreateMetadataOnlyKubeClient creates a Kubernetes clientset and a custom resource clientset.
+func CreateMetadataOnlyKubeClient(apiserver string, kubeconfig string) (metadata.Interface, error) {
+	if currentMetadataOnlyKubeClient != nil {
+		return currentMetadataOnlyKubeClient, nil
+	}
+
+	var err error
+
+	if config == nil {
+		var err error
+		config, err = clientcmd.BuildConfigFromFlags(apiserver, kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+	}
+	config.UserAgent = fmt.Sprintf("%s/%s (%s/%s) kubernetes/%s", "kube-state-metrics (metadataonly)", version.Version, runtime.GOOS, runtime.GOARCH, version.Revision)
+	config.AcceptContentTypes = "application/vnd.kubernetes.protobuf;as=PartialObjectMetadataList;g=meta.k8s.io;v=v1,application/json;as=PartialObjectMetadataList;g=meta.k8s.io;v=v1,application/json,application/vnd.kubernetes.protobuf;as=PartialObjectMetadata;g=meta.k8s.io;v=v1,application/json;as=PartialObjectMetadata;g=meta.k8s.io;v=v1"
```

Comment:
Do we really need to do that ourselves? The client seems to already set the Accept header accordingly: https://github.com/kubernetes/kubernetes/blob/0b3b733c8437a946b7300a346ac948fec2e0b3ff/staging/src/k8s.io/client-go/metadata/metadata.go#L222

Reply:
This was mainly copying from the CreateKubeClient() func, I'm not sure if we need it here additionally. 

Will metadata.List()/Watch() be called eventually or is that overwritten by ListWatchFunc?